packages/toast/src/queue.ts

```diff
@@ -1,14 +1,30 @@
 import { AddToastOptions } from "./types";
 
+type Subscriber = () => void;
+
 export class ToastQueue {
     private static queue: AddToastOptions[] = [];
-    private static isLocked = false;
-
-    public static lock = () => (this.isLocked = true);
-    public static unlock = () => (this.isLocked = false);
-    public static push = (item: AddToastOptions) => this.queue.push(item);
+    private static subscribers: Set<Subscriber> = new Set<Subscriber>();
+    private static emitChange = () => {
+        for (const subscriber of this.subscribers) {
+            subscriber();
+        }
+    };
+    public static subscribe = (subscriber: Subscriber) => {
+        this.subscribers.add(subscriber);
+        return () => {
+            this.subscribers.delete(subscriber);
+        };
+    };
+    public static getSnapshot = () => this.queue;
+    public static enqueue = (item: AddToastOptions) => {
+        this.queue = [...this.queue, item];
```

Comment:
I guess we need completely new objects for `useSyncExternalStore` to work?

Reply:
Yeah. Took me a while before I caught that initially.
They need to be fresh copies every time for the hook to detect store changes